ci/setup_env.sh

```diff
@@ -103,6 +81,46 @@ fi
 echo "activate pandas-dev"
 source activate pandas-dev
 
+echo
+echo "set the compiler cache to work"
+if [ -z "$NOCACHE" ] && [ "${TRAVIS_OS_NAME}" == "linux" ]; then
+    echo "Install ccache"
+    conda install ccache -y
+    echo "Using ccache"
+    ccache --show-stats
+    ccache --version
+    
+    if [ -z "$GCC" ]; then
+        GCC=$(which gcc)
+    fi
+    echo "gcc: $GCC"
+    CCACHE=ccache
+    echo "ccache: $CCACHE"
+    export CC="ccache $CC"
+elif [ -z "$NOCACHE" ] && [ "${TRAVIS_OS_NAME}" == "osx" ]; then
```

Comment:
Can this conditional branch be combined with the one above? Seems like mostly copy / paste save one off things so would be easier to read if consolidated